### ISSUE TITLE
diFFit plugin generates ImportError when wx not available

### ISSUE THREAD

**Issue by maurov** [xraypy/xraylarch#164]
Hello,

@newville if I run Larch as 'library only' mode, that is, without wx and other non-core dependencies, the diFFit plugin prevents the interpreter to instantiate, raising an ImportError (traceback below). Unfortunately, I did not understand how to correct this bug in 'builtins.py'. Other wx-based plugins also do 'import wx' but they are silent when wx is not available...

Here the traceback:

>>> from larch import Interpreter
>>> l = Interpreter()
  File "/home/mauro/local/py35/lib/python3.5/site-packages/larch/builtins.py", line 401, in _plugin_file
    out = imp.load_module(plugin, fh, modpath, desc)
  File "/usr/lib/python3.5/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.5/imp.py", line 172, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 693, in _load
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 665, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "/home/mauro/local/py35/share/larch/plugins/diFFit/ImageControlsFrame.py", line 8, in <module>
    import wx
Python Error at plugin '/home/mauro/local/py35/share/larch/plugins/diFFit/ImageControlsFrame.py', line 0
   ^
ImportError: 

Thanks for your help!

**Comment by newville** [comment]
@maurov  @margaretkolbeannellen  Ah, sorry for the trouble.   I think the best solution is to add a `requirements.txt` to `plugins/diFFit` that includes its dependencies, including wx.

**Comment by maurov** [comment]
@newville no problem. Probably plugins/diFFit could behave as plugins/wx: if wx is not available, the GUI part will not be available and stay silent. Please, could you explain how is realized that the 'import wx' in plugings/wx does not trigger the same ImportError?

**Comment by newville** [comment]
@maurov  @margaretkolbeannellen The file `plugins/wx/requirements.txt` is used to list the dependencies for the plugin.  Copying that to `plugins/diFFit/' would probably fix your immediate problem, and we should be more careful about adding that other plugins too.

**Comment by maurov** [comment]
@newville @margaretkolbeannellen Copying `plugins/wx/requirements.txt` does not fix the ImportError problem completely (you can see what I did in my `diffit_wx` branch) and I get it at another point. The traceback is below:

```
>>> from larch import Interpreter
>>> l = Interpreter()
  File "/home/mauro/local/py35/lib/python3.5/site-packages/larch/builtins.py", line 401, in _plugin_file
    out = imp.load_module(plugin, fh, modpath, desc)
  File "/usr/lib/python3.5/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.5/imp.py", line 172, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 693, in _load
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 665, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "/home/mauro/local/py35/share/larch/plugins/xrmmap/xrm_mapfile.py", line 17, in <module>
    from larch_plugins.xrmmap import (FastMapConfig, read_xrf_netcdf,
  File "/home/mauro/local/py35/share/larch/plugins/xrmmap/__init__.py", line 7, in <module>
    from .xrm_mapfile import (read_xrfmap, h5str,
  File "/home/mauro/local/py35/share/larch/plugins/xrmmap/xrm_mapfile.py", line 23, in <module>
    from larch_plugins.diFFit.xrd import XRD
  File "/home/mauro/local/py35/share/larch/plugins/diFFit/__init__.py", line 3, in <module>
    from .cifdb import cifDB
  File "/home/mauro/local/py35/share/larch/plugins/diFFit/cifdb.py", line 36, in <module>
    from larch_plugins.diFFit.XRDCalculations import generate_hkl
  File "/home/mauro/local/py35/share/larch/plugins/diFFit/XRDCalculations.py", line 26, in <module>
    import wx
Python Error at plugin '/home/mauro/local/py35/share/larch/plugins/xrmmap/xrm_mapfile.py', line 0
   ^
ImportError: 
```

PS: I'm using python 3

**Comment by maurov** [comment]
#166 plus the minor fixes for python3 solve this issue, thanks!

@margaretkolbeannellen please, could you check it and then pull request back in xraypy/xraylarch?

**Comment by maurov** [comment]
Fixed by #167, great!
